presto-main/src/main/java/com/facebook/presto/sql/planner/plan/DistinctLimitNode.java

```diff
@@ -74,18 +85,39 @@ public long getLimit()
         return hashSymbol;
     }
 
+    @JsonProperty("distinctSymbols")
     public List<Symbol> getDistinctSymbols()
     {
         if (hashSymbol.isPresent()) {
-            return ImmutableList.copyOf(Iterables.filter(getOutputSymbols(), not(hashSymbol.get()::equals)));
+            return ImmutableList.copyOf(Iterables.filter(distinctSymbols, not(hashSymbol.get()::equals)));
         }
-        return getOutputSymbols();
+        return distinctSymbols;
+    }
+
+    @JsonProperty("nonDistinctSymbols")
```

Comment:
Why do we need nonDistinctSymbols and aggregationSymbolMap in this class? they seem to be never used by anything.